.github/workflows/test.yaml

```diff
@@ -3,12 +3,30 @@ name: test
 on: pull_request
 
 jobs:
-  test:
+  tests:
+    name: Testing on python ${{ matrix.python_version }}
     runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        python_version: ['3.10', '3.11', '3.12']
+
     steps:
       - uses: actions/checkout@v2
 
       - uses: citizensadvice/python-poetry-setup-action@v1
+        with:
+          python_version: ${{ matrix.python_version }}
 
       - name: Run tests
-        run: poetry run pytest
+        run: poetry run pytest -vv
+
+  test:
+    # Single job for github to expect. Any merge-blocking checks should be added to the needs array
```

Comment:
would both `test` and `tests` be selectable branch protection settings? If yes then this could be confusing.

Reply:
no, it goes off the name element of the job (which does default to the key), so there's no `tests` available in the branch protection search, because it's not a job itself.